File: client/lua/krpc/test/test_performance.lua
local luaunit = require 'luaunit'
local class = require 'pl.class'
local ServerTest = require 'krpc.test.servertest'

local TestPerformance = class(ServerTest)

function TestPerformance:test_performance()
  local n = 100
  local function wrapper()
    self.conn.test_service.float_to_string(3.14159)
  end
  local t1 = os.clock()
  for i=1,n do
    wrapper()
  end
  local t2 = os.clock()
  local t = t2 - t1
  print()
  print(string.format('Total execution time: %.4f seconds', t))
  print(string.format('RPC execution rate: %d per second', (n/t)))
  print(string.format('Latency: %.3f milliseconds', ((t*1000)/n)))
end

return TestPerformance

File: client/python/krpc/test/test_performance.py
from __future__ import print_function
import unittest
import timeit
from krpc.test.servertestcase import ServerTestCase


class TestPerformance(ServerTestCase, unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        super(TestPerformance, cls).setUpClass()

    def test_performance(self):
        samples = 100

        def wrapper():
            self.conn.test_service.float_to_string(float(3.14159))

        delta_t = timeit.timeit(stmt=wrapper, number=samples)
        print()
        print('Total execution time: %.2f seconds' % delta_t)
        print('RPC execution rate: %d per second' % (samples/delta_t))
        print('Latency: %.3f milliseconds' % ((delta_t*1000)/samples))

if __name__ == '__main__':
    unittest.main()
